### DISCUSSION TITLE
What does the "installed" field mean in the cask JSON API?

### DISCUSSION THREAD

**Discussion by tddschn** [Homebrew/discussions#2080] [Everyday usage] [answered]
```bash
$ curl https://formulae.brew.sh/api/cask.json -s | jq '.[] | select(.installed != null) | .token'
"chromedriver"
"firefox"
"google-chrome"
"julia"
"microsoft-auto-update"
"microsoft-edge"
"powershell"
"r"
"session-manager-plugin"
"soundflower"
"vagrant"
"virtualbox"
```
This outputs these casks whose `installed` field aren't `null`. 

For `google-chrome`, it looks like this:
```
  {
    "token": "google-chrome",
    "full_token": "google-chrome",
    "tap": "homebrew/cask",
    "name": ["Google Chrome"],
    "desc": "Web browser",
    "homepage": "https://www.google.com/chrome/",
    "url": "https://dl.google.com/chrome/mac/stable/GGRO/googlechrome.dmg",
    "appcast": null,
    "version": "93.0.4577.63",
    "versions": {},
    "installed": "92.0.4515.131",
    "outdated": false,
    ...
```
Obviously the API endpoint has no knowledge of my local cask installations.
So what does the "installed" field mean in the cask JSON API?

**Comment by SMillerDev** [comment]
@Rylan12 probably knows best. But I think it's just some data of the CI machine leaking into the api.

**Comment by Rylan12** [comment, marked as answer]
@SMillerDev is correct, it's just info leaking through that shows which casks are installed on the CI machine that was used to generate the JSON API files.

The files that are generated almost exactly match the output of `brew info --json=v2 CASK`. If you run that locally, you will see the `installed` field be appropriately filled in.
